circuit_puzzles/oracle.clsp

```diff
@@ -28,20 +29,31 @@
     ; recreate yourself to conserve updates so far
     (verify-announcement-asserts
       input_announcement_asserts
-      (list
-        (list CREATE_COIN
-          (curry_hashes MOD_HASH
-            (sha256 ONE MOD_HASH)
-            (sha256tree STATUTES_STRUCT)
-            (sha256 ONE ATOM_ANNOUNCER_PUZZLE_HASH)
-            (sha256tree PRICES)
+      (assign
+        (current_timestamp price_expiration_seconds) announce_args
+        delayed_timestamp (if (all current_timestamp price_expiration_seconds) (- current_timestamp price_expiration_seconds) 0)
+        (last_matured_price . _) (cut-prices PRICES delayed_timestamp ())
+        (oldest_price . oldest_updated) (if last_matured_price last_matured_price (c () ()))
+        (list
+          (list CREATE_COIN
+            (curry_hashes MOD_HASH
+              (sha256 ONE MOD_HASH)
+              (sha256tree STATUTES_STRUCT)
+              (sha256 ONE ATOM_ANNOUNCER_PUZZLE_HASH)
+              (sha256tree PRICES)
+            )
+            ; min singleton amount, force it to be this
+            ONE
           )
-          ; min singleton amount, force it to be this
-          ONE
-        )
-        (assign
-          (oldest_price . oldest_updated) (f (cut-prices PRICES 0))
-          (list CREATE_PUZZLE_ANNOUNCEMENT (sha256 oldest_price oldest_updated))
+          (if oldest_updated
+            (list CREATE_PUZZLE_ANNOUNCEMENT (sha256 oldest_price oldest_updated current_timestamp price_expiration_seconds))
+            (x)  ; no matured prices available, fail, this only occurs on launch
```

Comment:
this can also happen if Statutes Price Delay (price_expiration_second) get increased by governance and un-matures the matured price